kolibri/plugins/learn/assets/src/state/actions/recommended.js

```diff
@@ -0,0 +1,280 @@
+import {
+  ContentNodeResource,
+  ContentNodeProgressResource,
+  SessionResource,
+  UserExamResource,
+  ExamLogResource,
+  ExamAttemptLogResource,
+} from 'kolibri.resources';
+
+import {
+  currentUserId,
+  isFacilityUser,
+  getCurrentChannelId,
+  getCurrentChannelObject,
+} from 'kolibri.coreVue.vuex.getters';
+
+import {
+  samePageCheckGenerator,
+  setChannelInfo,
+  handleApiError,
+} from 'kolibri.coreVue.vuex.actions';
+
+import { PageNames } from '../../constants';
+import { _contentState } from './main';
+
+import ConditionalPromise from 'kolibri.lib.conditionalPromise';
+import uniqBy from 'lodash/uniqBy';
+
+function _fetchPopular(channelId) {
+  const channelPayload = { channel_id: channelId };
+  const popularPayload = { popular: 'true' };
+  return ContentNodeResource.getCollection(channelPayload, popularPayload).fetch();
+}
+
+function _fetchNextSteps(channelId, state) {
+  const nextStepsPayload = { next_steps: currentUserId(state) };
+  const channelPayload = { channel_id: channelId };
+
+  if (isFacilityUser(state)) {
+    return ContentNodeResource.getCollection(channelPayload, nextStepsPayload).fetch();
+  }
+  return Promise.resolve([]);
+}
+
+function _fetchResume(channelId, state) {
+  const resumePayload = { resume: currentUserId(state) };
+  const channelPayload = { channel_id: channelId };
+
+  // all you really need is user_id. Should we check in parent?
+  if (isFacilityUser(state)) {
+    return ContentNodeResource.getCollection(channelPayload, resumePayload).fetch();
+  }
+  return Promise.resolve([]);
+}
+
+function _fetchOverview(channelId, cursor) {
```

Comment:
The size of the cursor batch is determined on the server side, so we would just need to change the endpoint. If there is no need to ever page to get subsequent items, we can get rid of the cursor (which is database and CPU intensive) and just get the first 40 items every time.